src/main/java/org/folio/inventory/dataimport/cache/MappingMetadataCache.java

```diff
@@ -0,0 +1,74 @@
+package org.folio.inventory.dataimport.cache;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.http.HttpStatus;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.inventory.dataimport.exceptions.CacheLoadingException;
+import org.folio.rest.client.MappingMetadataClient;
+import org.folio.rest.jaxrs.model.MappingMetadataDto;
+import org.folio.rest.util.OkapiConnectionParams;
+
+import com.github.benmanes.caffeine.cache.AsyncCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+
+/**
+ * Cache for storing MappingMetadataDto entities by jobExecutionId
+ */
+public class MappingMetadataCache {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  private final AsyncCache<String, Optional<MappingMetadataDto>> cache;
+  private final HttpClient httpClient;
+
+  public MappingMetadataCache(Vertx vertx, HttpClient httpClient, long cacheExpirationTime) {
+    this.httpClient = httpClient;
+    cache = Caffeine.newBuilder()
+      .expireAfterAccess(cacheExpirationTime, TimeUnit.SECONDS)
+      .executor(task -> vertx.runOnContext(v -> task.run()))
+      .buildAsync();
+  }
+
+  public Future<Optional<MappingMetadataDto>> get(String jobExecutionId, Map<String, String> context) {
+    try {
+      return Future.fromCompletionStage(cache.get(jobExecutionId, (key, executor) -> loadJobProfileSnapshot(key, context)));
+    } catch (Exception e) {
+      LOGGER.warn("Error loading MappingMetadata by jobExecutionId: '{}'", jobExecutionId, e);
+      return Future.failedFuture(e);
+    }
+  }
+
+  private CompletableFuture<Optional<MappingMetadataDto>> loadJobProfileSnapshot(String jobExecutionId, Map<String, String> context) {
+    LOGGER.debug("Trying to load MappingMetadata by jobExecutionId  '{}' for cache, okapi url: {}, tenantId: {}", jobExecutionId, context.get(OkapiConnectionParams.OKAPI_URL_HEADER), context.get(OkapiConnectionParams.OKAPI_TENANT_HEADER));
+
+    MappingMetadataClient client = new MappingMetadataClient(context.get(OkapiConnectionParams.OKAPI_URL_HEADER), context.get(OkapiConnectionParams.OKAPI_TENANT_HEADER), context.get(OkapiConnectionParams.OKAPI_TOKEN_HEADER), httpClient);
+
+    return client.getMappingMetadataByJobExecutionId(jobExecutionId)
```

Comment:
Does this make a HTTP request to a FOLIO interface to fetch the job profile information to be cached?

Reply:
@marcjohnson-kint  yes. It will load these data:
load JobProfileSnapshot from mod-data-import-converter-storage by HTTP
load MappingRules and Mapping Params from mod-source-record-manager by HTTP

Reply:
That suggests to me that new interface dependencies are needed for mod-inventory to reflect the new dependency on these APIs.

Have I understood this correctly?

Reply:
Yes, it seems like it will work in this way.

Reply:
Please make sure those interface requirements / dependencies are described in the module descriptor

Reply:
Thank you for your comment @marcjohnson-kint !
I added the needed interfaces.